app/app/src/main/java/com/example/myapplication/MainActivity.java

```diff
@@ -468,4 +483,25 @@ public void run() {
         Thread animationInteractionThread = new Thread(animationInteraction);
         animationInteractionThread.start();
     }
-}
\ No newline at end of file
+
+    public void connectionHelper(final String name) {
+        Runnable animationInteraction = new Runnable() {
+            @Override
+            public void run() {
+                Process.setThreadPriority(Process.THREAD_PRIORITY_MORE_FAVORABLE);
+                if(name.equals("Car")){
+                    Car.findBT(name);
+                    try {
+                        Car.openBT();
+                    } catch (IOException e) {
+                        e.printStackTrace();
+                    }
+                } else
+                    Headset.findBT(name);
+            }
+        };
+
+        Thread animationInteractionThread = new Thread(animationInteraction);
```

Comment:
The Thread should be named something that relates to the situation, `animationInteractionThread` makes it sound like it relates to animation.

Reply:
fixed